zcash_primitives/src/transaction/mod.rs

```diff
@@ -544,22 +561,20 @@ pub mod testing {
         }
     }
 
-    fn tx_versions(branch_id: BranchId) -> impl Strategy<Value = (u32, u32)> {
+    #[cfg(feature = "zfuture")]
+    fn tx_versions(branch_id: BranchId) -> impl Strategy<Value = TxVersion> {
         match branch_id {
-            BranchId::Sprout => (1..(2 as u32)).prop_map(|i| (i, 0)).boxed(),
-            BranchId::Overwinter => {
-                Just((OVERWINTER_TX_VERSION, OVERWINTER_VERSION_GROUP_ID)).boxed()
-            }
-            #[cfg(feature = "zfuture")]
-            BranchId::ZFuture => Just((ZFUTURE_TX_VERSION, ZFUTURE_VERSION_GROUP_ID)).boxed(),
-            _otherwise => Just((SAPLING_TX_VERSION, SAPLING_VERSION_GROUP_ID)).boxed(),
+            BranchId::Sprout => (1..(3 as u32)).prop_map(|i| TxVersion::Sprout(i)).boxed(),
```

Comment:
I think this is clearer for conveying the expected range:
```suggestion
            BranchId::Sprout => (1..=2u32).prop_map(|i| TxVersion::Sprout(i)).boxed(),
```